web-ui/src/context/selectors.js

```diff
@@ -128,6 +130,62 @@ export const selectHasUpdateReviewPeriodPermission = hasPermission(
   'CAN_UPDATE_REVIEW_PERIOD'
 );
 
+export const selectHasMeritReportPermission = hasPermission(
+  'CAN_CREATE_MERIT_REPORT'
+);
+
+export const selectHasEarnedCertificationsPermission = hasPermission(
+  'CAN_MANAGE_EARNED_CERTIFICATIONS'
+);
+
+export const selectHasVolunteeringEventsPermission = hasPermission(
+  'CAN_ADMINISTER_VOLUNTEERING_EVENTS'
+);
+
+export const selectHasVolunteeringOrganizationsPermission = hasPermission(
+  'CAN_ADMINISTER_VOLUNTEERING_ORGANIZATIONS'
+);
+
+export const selectHasVolunteeringRelationshipsPermission = hasPermission(
+  'CAN_ADMINISTER_VOLUNTEERING_RELATIONSHIPS'
+);
+
+export const selectHasRoleAssignmentPermission = hasPermission(
+  'CAN_ASSIGN_ROLE_PERMISSIONS'
+);
+
+export const selectHasCreateFeedbackPermission = hasPermission(
+  'CAN_CREATE_FEEDBACK_REQUEST'
+);
+
+export const selectHasAdministerKudosPermission = hasPermission(
+  'CAN_ADMINISTER_KUDOS'
+);
+
+export const selectHasDeleteMembersPermission = hasPermission(
+  'CAN_DELETE_ORGANIZATION_MEMBERS'
+);
+
+export const selectHasCreateMembersPermission = hasPermission(
+  'CAN_CREATE_ORGANIZATION_MEMBERS'
+);
+
+export const selectHasImpersonateMembersPermission = hasPermission(
+  'CAN_IMPERSONATE_MEMBERS'
+);
+
+export const selectHasViewSettingsPermission = hasPermission(
+  'CAN_VIEW_SETTINGS'
+);
+
+export const selectHasAdministerSettingsPermission = hasPermission(
+  'CAN_ADMINISTER_SETTINGS'
+);
+
+export const selectHasSendEmailPermission = hasPermission(
+  'CAN_SEND_EMAIL'
+);
+
```

Comment:
```
const permissionKeys = {
  selectHasMeritReportPermission: 'CAN_CREATE_MERIT_REPORT',
  selectHasEarnedCertificationsPermission: 'CAN_MANAGE_EARNED_CERTIFICATIONS',
  selectHasVolunteeringEventsPermission: 'CAN_ADMINISTER_VOLUNTEERING_EVENTS',
  selectHasVolunteeringOrganizationsPermission: 'CAN_ADMINISTER_VOLUNTEERING_ORGANIZATIONS',
  selectHasVolunteeringRelationshipsPermission: 'CAN_ADMINISTER_VOLUNTEERING_RELATIONSHIPS',
  selectHasRoleAssignmentPermission: 'CAN_ASSIGN_ROLE_PERMISSIONS',
  selectHasCreateFeedbackPermission: 'CAN_CREATE_FEEDBACK_REQUEST',
  selectHasAdministerKudosPermission: 'CAN_ADMINISTER_KUDOS',
  selectHasDeleteMembersPermission: 'CAN_DELETE_ORGANIZATION_MEMBERS',
  selectHasCreateMembersPermission: 'CAN_CREATE_ORGANIZATION_MEMBERS',
  selectHasImpersonateMembersPermission: 'CAN_IMPERSONATE_MEMBERS',
  selectHasViewSettingsPermission: 'CAN_VIEW_SETTINGS',
  selectHasAdministerSettingsPermission: 'CAN_ADMINISTER_SETTINGS',
  selectHasSendEmailPermission: 'CAN_SEND_EMAIL'
};

const createPermissionSelectors = (permissionsMap) =>
  Object.entries(permissionsMap).reduce((selectors, [key, permission]) => {
    selectors[key] = hasPermission(permission);
    return selectors;
  }, {});

export const {
  selectHasMeritReportPermission,
  selectHasEarnedCertificationsPermission,
  selectHasVolunteeringEventsPermission,
  selectHasVolunteeringOrganizationsPermission,
  selectHasVolunteeringRelationshipsPermission,
  selectHasRoleAssignmentPermission,
  selectHasCreateFeedbackPermission,
  selectHasAdministerKudosPermission,
  selectHasDeleteMembersPermission,
  selectHasCreateMembersPermission,
  selectHasImpersonateMembersPermission,
  selectHasViewSettingsPermission,
  selectHasAdministerSettingsPermission,
  selectHasSendEmailPermission,
} = createPermissionSelectors(permissionKeys);
```

Reply:
While this suggestion will result in fewer lines and less repetition, I think it's less readable. Someone coming in trying to grok what one of these selectors is doing first has to understand what's going on with the selector generation.

I'm 50/50 on this one.

Reply:
What do you think of this?

https://gist.github.com/thelenw/0f6ef533ca4b1dc23e1234f767a34460